plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -32,81 +29,82 @@ public function watchFolder(KalturaDropFolder $dropFolder)
 		/* @var $dropFolder KalturaMicrosoftTeamsDropFolder */
 		$this->dropFolder = $dropFolder;
 		$this->vendorPlugin = KalturaVendorClientPlugin::get(KBatchBase::$kClient);
-		$this->graphClient = new KMicrosoftGraphClient($dropFolder->tenantId, $dropFolder->path, $dropFolder->clientId, $dropFolder->clientSecret);
 
 		$vendorIntegrationSetting = $this->vendorPlugin->vendorIntegration->get($dropFolder->integrationId);
 		KalturaLog::info('Watching folder ['.$this->dropFolder->id.']');
-		$driveLocationsDir =  KBatchBase::$taskConfig->params->teams->driveLocationsDir;
-		$driveLocationsFileNamePrefix = KBatchBase::$taskConfig->params->teams->driveLocationsFileNamePrefix;
 
-		$existingDropFolderFiles = $this->loadDropFolderFiles();
+        $existingDropFolderFiles = $this->loadDropFolderFiles();
 
-		$files = scandir($driveLocationsDir);
+		// Retrieve all user custom metadata objects for the metadata profile specified on the drop folder's integration ID
+        $userMetadataSearchItem = new KalturaMetadataSearchItem();
+        $userMetadataSearchItem->metadataProfileId = $vendorIntegrationSetting->userMetadataProfileId;
+        $userMetadataSearchItem->orderBy = self::ASC_LAST_ACCESSED_DATE;
 
-		$currentFile = null;
-		foreach ($files as $file)
-		{
-			if ($file == '.' || $file == '..' || strpos($file,self::LOCKED_FILE_MIDFIX) || strpos($file, $driveLocationsFileNamePrefix . $this->dropFolder->id) === false)
-			{
-				continue;
-			}
-			else
-			{
-				$currentFile = $file;
-				break;
-			}
-		}
-
-		if (!$currentFile)
-		{
-			KalturaLog::info('No iterable file found. Skipping.');
-			return;
-		}
-
-		KalturaLog::info("On this iteration, processing file $currentFile");
+        $condition = new KalturaSearchMatchCondition();
+        $condition->field = self::OPT_IN_XPATH;
+        $condition->value = self::OPT_IN_VALUE;
 
-		$currentFilePieces = explode('.', $currentFile, 2);
-		$lockedFileName = $currentFilePieces[0] . self::LOCKED_FILE_MIDFIX . $currentFilePieces[1];
+        $userMetadataSearchItem->items = [ $condition ];
 
-		rename($driveLocationsDir . DIRECTORY_SEPARATOR . $currentFile, $driveLocationsDir . DIRECTORY_SEPARATOR . $lockedFileName);
+        $userFilter = new KalturaUserFilter();
+        $userFilter->advancedSearch = $userMetadataSearchItem;
 
-		$itemCount = 0;
+        $pager = new KalturaFilterPager();
+        $pager->pageIndex = 1;
+        $pager->pageSize = KBatchBase::$taskConfig->params->teams->userPageSize;
 
-		$drivesFileHandle = fopen($driveLocationsDir . DIRECTORY_SEPARATOR . $lockedFileName, 'r+');
+        KBatchBase::$kClient->setResponseProfile($this->constructResponseProfile($vendorIntegrationSetting->userMetadataProfileId));
 
-		$newDriveUrlAssoc = array();
-		$line = fgetcsv($drivesFileHandle);
+        $response = KBatchBase::$kClient->user->listAction($userFilter, $pager);
```

Comment:
better have it in try catch

Reply:
@inbal-ben-david  list requests cannot throw exceptions unless the KS is invalid.